Building VerneMQ in macOS Catalina

Hi there, mac user here.

### Environment

- VerneMQ Version: from master
- OS: macOS Catalina version 10.15.1
- Erlang/OTP version (if building from source): 22.1.8

### Expected behavior

Being able to follow the [tutorial](https://github.com/vernemq/vernemq/wiki/Building-VerneMQ-on-OS-X) on how to build on OS-X and build VerneMQ.

### Actual behaviour

It is not possible to install the Xcode CLT (Command Line Tools) 9.4.1 on Catalina so I don't know how to get around the issues compiling the leveldb package.

```bash
.
.
.
1 warning and 1 error generated.
cc  -I /Users/thales.biancalana/Desktop/vernemq/_build/default/lib/eleveldb/c_src/system/include -I. -I./include -mmacosx-version-min=10.8 -DOS_MACOSX -DLEVELDB_PLATFORM_POSIX -DSNAPPY  -O2 -g -DNDEBUG     -fPIC -MM -E -MT db/c.d -MT db/c.o -MF db/c.d db/c.cc
warning: include path for stdlibc++ headers not found; pass '-stdlib=libc++' on the command line to use the libc++ standard library instead [-Wstdlibcxx-not-found]
In file included from db/c.cc:9:
In file included from ./include/leveldb/cache.h:22:
./include/leveldb/slice.h:21:10: fatal error: 'string' file not found
#include <string>
         ^~~~~~~~
1 warning and 1 error generated.
cc  -I /Users/thales.biancalana/Desktop/vernemq/_build/default/lib/eleveldb/c_src/system/include -I. -I./include -mmacosx-version-min=10.8 -DOS_MACOSX -DLEVELDB_PLATFORM_POSIX -DSNAPPY  -O2 -g -DNDEBUG     -fPIC -MM -E -MT db/builder.d -MT db/builder.o -MF db/builder.d db/builder.cc
warning: include path for stdlibc++ headers not found; pass '-stdlib=libc++' on the command line to use the libc++ standard library instead [-Wstdlibcxx-not-found]
In file included from db/builder.cc:8:
In file included from ./db/builder.h:8:
./include/leveldb/status.h:16:10: fatal error: 'string' file not found
#include <string>
         ^~~~~~~~
1 warning and 1 error generated.
g++  -I /Users/thales.biancalana/Desktop/vernemq/_build/default/lib/eleveldb/c_src/system/include -I. -I./include -mmacosx-version-min=10.8 -DOS_MACOSX -DLEVELDB_PLATFORM_POSIX -DSNAPPY  -O2 -g -DNDEBUG     -fPIC -c db/builder.cc -o db/builder.o
warning: include path for stdlibc++ headers not found; pass '-stdlib=libc++' on the command line to use the libc++ standard library instead [-Wstdlibcxx-not-found]
In file included from db/builder.cc:8:
In file included from ./db/builder.h:8:
./include/leveldb/status.h:16:10: fatal error: 'string' file not found
#include <string>
         ^~~~~~~~
1 warning and 1 error generated.
make[1]: *** [db/builder.o] Error 1
```

--- a/.gitignore
+++ b/.gitignore
@@ -5,6 +5,7 @@ vars.generated
 vars/dev*_vars.config
 *.deb
 *.rpm
+*.crashdump
 
 .ropeproject
 *~

--- a/.travis.yml
+++ b/.travis.yml
@@ -7,6 +7,7 @@ services:
   - memcached
 
 before_install:
+  - sudo apt-get -y install libsnappy-dev
   - mysql -e "CREATE DATABASE vmq_test_database;" -uroot
   - mysql -e "CREATE USER 'vmq_test_user' IDENTIFIED BY 'vmq_test_password';" -uroot
   - mysql -e "GRANT ALL PRIVILEGES ON * . * TO 'vmq_test_user';" -uroot

--- a/apps/vmq_generic_msg_store/rebar.config
+++ b/apps/vmq_generic_msg_store/rebar.config
@@ -1,7 +1,7 @@
 {erl_opts, [debug_info, {parse_transform, lager_transform}]}.
 {deps, [
         lager,
-        {eleveldb, {git, "git://github.com/vernemq/eleveldb.git", "develop"}},
+        {eleveldb, {git, "git://github.com/vernemq/eleveldb.git", {branch, "develop"}}},
         {sext, "1.5.0"}
        ]}.
 

--- a/apps/vmq_swc/rebar.config
+++ b/apps/vmq_swc/rebar.config
@@ -3,7 +3,7 @@
         lager,
         {sext, "1.5.0"},
         {swc, {git, "git://github.com/vernemq/ServerWideClocks.git", "master"}},
-        {eleveldb, {git, "git://github.com/vernemq/eleveldb.git", "develop"}},
+        {eleveldb, {git, "git://github.com/vernemq/eleveldb.git", {branch, "develop"}}},
         riak_dt
        ]}.
 

--- a/changelog.md
+++ b/changelog.md
@@ -6,6 +6,7 @@
 - Fix bug where vmq_metrics crashes because external metric providers haven't
   started yet.
 - Fix bug in vmq_swc where a cluster leave didn't properly cleanup the node clock.
+- Fix compilation on OSX Mojave and Catalina. Requires running `brew install openssl snappy`.
 
 ## VerneMQ 1.10.0
 

--- a/rebar.lock
+++ b/rebar.lock
@@ -29,7 +29,7 @@
   1},
  {<<"eleveldb">>,
   {git,"git://github.com/vernemq/eleveldb.git",
-       {ref,"751961c905764ef947330773d6678cac5991d98f"}},
+       {ref,"06fefb8102b7fd344c3c56810dee8a2f2c9691d7"}},
   0},
  {<<"emysql">>,
   {git,"git://github.com/djustinek/Emysql.git",

